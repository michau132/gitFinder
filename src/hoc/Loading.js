import React from 'react';
import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

const ldsHourglass = keyframes`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`;

export const WithLoadingStyles = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: inline-block;
  width: 192px;
  height: 192px;
  
  ::after {
    content: " ";
  display: block;
  width: 136px;
  height: 136px;
  margin: 1px;
  border-radius: 50%;
  border: 5px solid rgb(70, 42, 42);
  border-color: rgb(70, 42, 42) transparent rgb(70, 42, 42) transparent;
  animation: ${ldsHourglass} 1.2s linear infinite;
  }
`;

const Loading = WrappedComponent => (props) => {
  const { isLoading } = props;
  if (isLoading) {
    return <WithLoadingStyles />;
  }
  return <WrappedComponent {...props} />;
};

Loading.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};


export default Loading;
