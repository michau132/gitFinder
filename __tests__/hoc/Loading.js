import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../../src/hoc/Loading';

const component = () => <h1>Hi there</h1>;
const ConditionalHOC = Loading(component);
const props = {
  restStore: {
    isLoading: true,
  },
};

describe('testing Loading HOC', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConditionalHOC {...props} />);
  });
  test('should render component when loading is set to true', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
