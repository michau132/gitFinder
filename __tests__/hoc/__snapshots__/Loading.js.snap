// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Loading HOC should match snapshot 1`] = `
<component
  restStore={
    Object {
      "isLoading": true,
    }
  }
/>
`;
